src/docs/cookbook/navigation/navigate-with-arguments.md

```diff
@@ -0,0 +1,332 @@
+---
+title: Navigate to a named route with arguments
+prev:
+  title: Animating a Widget across screens
+  path: /docs/cookbook/navigation/hero-animations
+next:
+  title: Navigate to a new screen and back
+  path: /docs/cookbook/navigation/navigation-basics
```

Comment:
About ordering: It feels like "navigate with arguments" is more specific than "navigation basics". Shouldn't the basics come before this new entry?

Reply:
Oh the curse of alphabetical ordering! 

I'll try to come up with a different title that makes sense and is alphabetically below "Navigation Basics"

Reply:
Renamed to "Pass arguments to a named route," which comes directly after "Navigate with named routes."

I'm not sure how long this balancing act will work, haha, but the problem is fixed for this PR.